src/bdk.udl

```diff
@@ -3,6 +3,8 @@ namespace bdk {
   ExtendedKeyInfo generate_extended_key(Network network, WordCount word_count, string? password);
   [Throws=BdkError]
   ExtendedKeyInfo restore_extended_key(Network network, string mnemonic, string? password);
+  [Throws=BdkError]
+  ChildKeyPair derive_child_key_pair(string str_xprv, string str_path);
```

Comment:
Do we not need to expose `ExtendedPrivKey` and `DerivationPath`? @notmandatory

Reply:
@notmandatory if this is the only way to create a child key pair, can it be a constructor?

Reply:
I modified the bdk-ff commit that the bdk-kotlin uses, and made same changes. After that, I added test for the deriveChildKeyPair() function 
```
@Test
    fun validChildKeys() {
        val str_xprv = "tprv8ZgxMBicQKsPdMTRB73kkTfC665bK3JL33eAFZ3MUiGZj8jniKQroG2yuamoQcRb24GG8dKRmVgaisjcZrDc9L3fyMf2NW19oajw8iWSwMV"
        val str_path = "m/84'/1'/0'/0"
        val childKeyPair = deriveChildKeyPair(str_xprv, str_path)
        assertEquals(
            "[59a3697d/84'/1'/0'/0]tprv8hZWYaWfKprzVnSVaaxwYun2yVp9TgpvcosjrGYwgL9hKMU1Jsrp3xKsNCpxSNBSEgUPp7c8A1MYLQcMcq7Ab9eafzNZUHQaBtGBvfCQiMg/*",
            childKeyPair.xprv
        )
        assertEquals(
            "[59a3697d/84'/1'/0'/0]tpubDEFYgzYuUCYfPFUHUEdXxKS9YXL5d21qC7UX8nbF6bx69qimwGgQESwjYN52awKwDd5PWGKXWAw27EBUk6QzSK3eEwsK8odY8FxVn4poRDp/*",
            childKeyPair.xpub
        )
    }
```

And It worked 

(I used the CLI to derive this result first, and used the same key for testing)

Reply:
Not discounting that it works, just checking if the types are okay or if we need something more elaborate.  and if child key pair could have a constructor instead of this function `deriveChildKeyPair`. also wondering if `ChildKeyPair` is what we want to expose - it could be any key pair, not necessarily only a "child" key pair. just thinking out loud.

Reply:
I think it's fine to keep the types and design for this simple to start and not add additional types for `ExtendedPrivKey` and `DerivationPath`. They can always be added later if we find a need, I think most apps will 1. `generate_extended_key` and get the user to write down the mnemonic, 2) from the `ExtendedKeyInfo.xprv` or possibly from importing a `xprv` from somewhere else,  derive a descriptor key, and 3) create a descriptor for the wallet, currently needs to be done putting strings together but in future we can make it simpler with some standard/common templates. That all said, I have two naming change suggestions:

1. rename `ChildKeyPair` to  `DescriptorKeyInfo` which is more in line with what `bdk` calls it
2. rename arguments `str_xprv` and `str_path` to simply `xprv` and `path` since type is already enforced thus redundant in the variable name.

As for creating the `ChildKeyPair` or `DescriptorKeyInfo` with a constructor, if we do that we need to change it from `dictionary` (pass by value) to an `interface` (pass by reference), so I'd prefer to keep it a `dictionary` and not use a constructor.

Reply:
Noted. Will force push the changes soon.